selectrum.el

```diff
@@ -2075,6 +2097,74 @@ Only to be used from `selectrum-select-from-history'"
          (propertize (selectrum-get-current-candidate 'notfull)
                      'selectum--insert t)))
 
+(defun selectrum--quick-keys (len keys)
+  "Get list of key combinations up to key length LEN.
+KEYS is a list of key strings to combine."
+  (unless (zerop len)
+    (cl-loop with list = keys
+             with olist = keys
+             repeat (1- len)
+             do (setq olist
+                      (cl-loop
+                       for char in list
+                       nconc (cl-loop for ochar in olist
+                                      collect (concat char ochar))))
+             finally return olist)))
+
+(defun selectrum--quick-read ()
+  "Read an index using `selectrum-quick-keys'."
+  (let* ((akeys (mapcar #'char-to-string selectrum-quick-keys))
+         (nkeys (length selectrum-quick-keys))
+         (needed selectrum--actual-num-candidates-displayed)
+         (len (ceiling (log needed nkeys)))
+         (keys (seq-take (selectrum--quick-keys len akeys) needed))
+         (input nil)
+         (read-key (lambda ()
+                     (unwind-protect (char-to-string (read-char))
+                       (let ((selectrum--quick-fun nil))
+                         (selectrum--update)))))
+         (selectrum--quick-fun
+          (lambda (i cand)
+            (let ((str (propertize (or (nth i keys) "")
+                                   'face 'selectrum-quick-keys-highlight)))
+              (when (and input (string-match (concat "\\`" input) str))
+                (setq str (copy-sequence str))
+                (add-face-text-property 0 (match-end 0) 'match t str))
+              (let ((start (min (length cand)
+                                (+ (length str)
+                                   (or (next-single-property-change
```

Comment:
Instead of handling the invisible properties here, we should just remove them already in `selectrum--display-string`. We handle 'display there, right? For this reason 'display just works.

Reply:
Yes sounds good :+1: